brainiak/utils/fmrisim.py

```diff
@@ -817,30 +841,27 @@ def convolve_hrf(stimfunction,
     # Create signal functions for each list in the stimfunction
     for list_counter in range(list_num):
 
-        # Take the stim function
-        stimfunction_temp = stimfunction[:, list_counter]
-
-        signal_function_temp = np.convolve(stimfunction_temp, hrf)
+        # Perform the convolution
+        signal_temp = np.convolve(stimfunction[:, list_counter], hrf)
 
-        # Decimate the signal function so that it only has one element per TR
-        decimate_interval = int(tr_duration * temporal_resolution)
-        signal_function_temp = signal_function_temp[0::decimate_interval]
-
-        # Cut off the HRF
-        last_timepoint = stimfunction_temp.shape[0] / tr_duration
-        last_timepoint /= temporal_resolution
-        signal_function_temp = signal_function_temp[0:int(last_timepoint)]
+        # Down sample the stim function so that it only has one element per
+        # TR. This accelerates the convolution greatly
+        signal_vox = np.zeros((duration,))
+        for sample in list(range(duration)):
```

Comment:
I think the suggestion in this post might do the same as the for loop but may be more efficient.
https://stackoverflow.com/questions/15956309/averaging-over-every-n-elements-of-a-numpy-array
 Just be careful to make sure that the `reshape` works, i.e., the length of `signal_vox` is dividable by `stride`